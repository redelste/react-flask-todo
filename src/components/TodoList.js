import React from 'react';
import Todo from './Todo';

import { makeStyles } from '@material-ui/core/styles';


import Draggable, { DraggableCore } from 'react-draggable'; // Both at the same time

// import { red } from '@material-ui/core/colors';

// we destructure props into the syntax {todos}.
// props are sent to us from App.js


const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
}));


const TodoList = ({ todos, setTodos }) => {
    const classes = useStyles();
    return (
        // data-testid gives the test a way to find the element I want to make assertions against 
        <div className={classes.root} data-testid="todo-list">
            {/* accessing the todos from the fetch, passed from the Todo component.  */}

            {todos
                .slice()
                .sort((todo1, todo2) => {
                    console.log(todo1)
                    return todo1.id - todo2.id

                })
                .map(todo =>
                    <Draggable
                        // axis="x"
                        // handle=".handle"
                        defaultPosition={{ x: 0, y: 0 }}
                        position={null}
                        // grid={[25, 25]}
                        scale={1}>
                        <div>
                            <Todo key={todo.id} todo={todo} todos={todos} setTodos={setTodos}></Todo>

                        </div>
                    </Draggable>
                )}


        </div>
    )
}
// }

export default TodoList;
