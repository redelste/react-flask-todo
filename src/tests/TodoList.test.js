import React from 'react';
import  TodoList  from "../components/TodoList";
import renderer from 'react-test-renderer';

let todo = {datecreated:"2020-08-06T13:56:00", description:"testing", id: 399, name: "a test to "}


let todos =[
    {datecreated:"2020-08-06T13:56:00", description:"testing", id: 399, name: "a test test the jesting"},
    {datecreated:"2020-08-06T14:29:00", description:"testing2", id: 400, name: "a test to test testing"},
    {datecreated:"2020-08-06T15:10:00", description:"testing3", id: 401, name: "a test to jest the jesting"}
]
test('TodoList renders correctly', ()=>{
    const component = renderer.create(<TodoList todos={todos}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})