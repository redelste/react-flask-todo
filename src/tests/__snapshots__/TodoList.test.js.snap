// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders correctly 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="todo-list"
>
  <div
    className={
      Object {
        "cancelIcon": "makeStyles-cancelIcon-11",
        "card": "makeStyles-card-3",
        "content": "makeStyles-content-4",
        "divider": "makeStyles-divider-5",
        "expand": "makeStyles-expand-9",
        "expandOpen": "makeStyles-expandOpen-10",
        "heading": "makeStyles-heading-6",
        "root": "makeStyles-root-2",
        "shapeCircle": "makeStyles-shapeCircle-8",
        "subheading": "makeStyles-subheading-7",
      }
    }
    data-testid="single-todo"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-cancelIcon-11"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
      <div
        className="MuiCardContent-root makeStyles-content-4"
      >
        <h6
          className="MuiTypography-root MuiTypography--heading MuiTypography-h6 MuiTypography-gutterBottom"
        >
          a test test the jesting
        </h6>
        <hr
          className="MuiDivider-root makeStyles-divider-5 MuiDivider-light"
        />
        <span
          className="MuiTypography-root MuiTypography--heading MuiTypography-caption MuiTypography-gutterBottom"
        >
          2020-08-06T13:56:00
        </span>
        <span
          className="MuiTypography-root MuiTypography--subheading MuiTypography-caption"
        >
          <hr
            className="MuiDivider-root makeStyles-divider-5 MuiDivider-light"
          />
        </span>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          testing1
        </p>
      </div>
      <div
        className="MuiCardActions-root"
      >
        <button
          aria-expanded={false}
          aria-label="show more"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-9"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className={
      Object {
        "cancelIcon": "makeStyles-cancelIcon-11",
        "card": "makeStyles-card-3",
        "content": "makeStyles-content-4",
        "divider": "makeStyles-divider-5",
        "expand": "makeStyles-expand-9",
        "expandOpen": "makeStyles-expandOpen-10",
        "heading": "makeStyles-heading-6",
        "root": "makeStyles-root-2",
        "shapeCircle": "makeStyles-shapeCircle-8",
        "subheading": "makeStyles-subheading-7",
      }
    }
    data-testid="single-todo"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-cancelIcon-11"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
      <div
        className="MuiCardContent-root makeStyles-content-4"
      >
        <h6
          className="MuiTypography-root MuiTypography--heading MuiTypography-h6 MuiTypography-gutterBottom"
        >
          a test to test testing
        </h6>
        <hr
          className="MuiDivider-root makeStyles-divider-5 MuiDivider-light"
        />
        <span
          className="MuiTypography-root MuiTypography--heading MuiTypography-caption MuiTypography-gutterBottom"
        >
          2020-08-06T14:29:00
        </span>
        <span
          className="MuiTypography-root MuiTypography--subheading MuiTypography-caption"
        >
          <hr
            className="MuiDivider-root makeStyles-divider-5 MuiDivider-light"
          />
        </span>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          testing2
        </p>
      </div>
      <div
        className="MuiCardActions-root"
      >
        <button
          aria-expanded={false}
          aria-label="show more"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-9"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className={
      Object {
        "cancelIcon": "makeStyles-cancelIcon-11",
        "card": "makeStyles-card-3",
        "content": "makeStyles-content-4",
        "divider": "makeStyles-divider-5",
        "expand": "makeStyles-expand-9",
        "expandOpen": "makeStyles-expandOpen-10",
        "heading": "makeStyles-heading-6",
        "root": "makeStyles-root-2",
        "shapeCircle": "makeStyles-shapeCircle-8",
        "subheading": "makeStyles-subheading-7",
      }
    }
    data-testid="single-todo"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-cancelIcon-11"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
      <div
        className="MuiCardContent-root makeStyles-content-4"
      >
        <h6
          className="MuiTypography-root MuiTypography--heading MuiTypography-h6 MuiTypography-gutterBottom"
        >
          a test to jest the jesting
        </h6>
        <hr
          className="MuiDivider-root makeStyles-divider-5 MuiDivider-light"
        />
        <span
          className="MuiTypography-root MuiTypography--heading MuiTypography-caption MuiTypography-gutterBottom"
        >
          2020-08-06T15:10:00
        </span>
        <span
          className="MuiTypography-root MuiTypography--subheading MuiTypography-caption"
        >
          <hr
            className="MuiDivider-root makeStyles-divider-5 MuiDivider-light"
          />
        </span>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          testing3
        </p>
      </div>
      <div
        className="MuiCardActions-root"
      >
        <button
          aria-expanded={false}
          aria-label="show more"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-9"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
